src/test/java/org/folio/rest/impl/LocationUnitLibraryIT.java

```diff
@@ -0,0 +1,205 @@
+
+@ExtendWith(VertxExtension.class)
+class LocationUnitLibraryIT
+  extends BaseReferenceDataIntegrationTest<Loclib, Loclibs> {
+
+  private String campusId;
+
+  @Override
+  protected String referenceTable() {
+    return LIBRARY_TABLE;
+  }
+
+  @Override
+  protected String resourceUrl() {
+    return "/location-units/libraries";
+  }
+
+  @Override
+  protected Class<Loclib> targetClass() {
+    return Loclib.class;
+  }
+
+  @Override
+  protected Class<Loclibs> collectionClass() {
+    return Loclibs.class;
+  }
+
+  @Override
+  protected Loclib sampleRecord() {
+    return new Loclib()
+      .withName("test-library")
+      .withCode("code")
+      .withCampusId(campusId);
+  }
+
+  @Override
+  protected Function<Loclibs, List<Loclib>> collectionRecordsExtractor() {
+    return Loclibs::getLoclibs;
+  }
+
+  @Override
+  protected List<Function<Loclib, Object>> recordFieldExtractors() {
+    return List.of(Loclib::getName);
+  }
+
+  @Override
+  protected Function<Loclib, String> idExtractor() {
+    return Loclib::getId;
+  }
+
+  @Override
+  protected Function<Loclib, Metadata> metadataExtractor() {
+    return Loclib::getMetadata;
+  }
+
+  @Override
+  protected UnaryOperator<Loclib> recordModifyingFunction() {
+    return classificationType -> classificationType.withName("name-updated");
+  }
+
+  @Override
+  protected List<String> queries() {
+    return List.of("name==test-library", "code=code", "campusId==" + campusId);
+  }
+
+  @BeforeEach
+  void beforeEach(Vertx vertx, VertxTestContext ctx) {
+    var postgresClient = PostgresClient.getInstance(vertx, TENANT_ID);
+    var institution = new Locinst().withName("institution").withCode("ic");
+    var campus = new Loccamp().withName("campus").withCode("cc");
+    postgresClient.save(INSTITUTION_TABLE, institution)
+      .compose(
+        id -> postgresClient.save(CAMPUS_TABLE, campus.withInstitutionId(id)))
+      .onFailure(ctx::failNow)
+      .onSuccess(id -> {
+        campusId = id;
+        ctx.completeNow();
+      });
+  }
+
+  @AfterEach
+  void afterEach(Vertx vertx, VertxTestContext ctx) {
+    var postgresClient = PostgresClient.getInstance(vertx, TENANT_ID);
+    postgresClient.delete(referenceTable(), (CQLWrapper) null)
+      .compose(rows -> postgresClient.delete(CAMPUS_TABLE, (CQLWrapper) null))
+      .compose(
+        rows -> postgresClient.delete(INSTITUTION_TABLE, (CQLWrapper) null))
+      .onFailure(ctx::failNow)
+      .onComplete(event -> ctx.completeNow());
+  }
+
+  @Test
+  void post_shouldReturn422_whenObjectIsDuplicate(Vertx vertx,
+                                                  VertxTestContext ctx) {
+    HttpClient client = vertx.createHttpClient();
+    String libraryId = UUID.randomUUID().toString();
+    var library = sampleRecord().withId(libraryId);
+
+    doPost(client, resourceUrl(), pojo2JsonObject(library))
+      .onComplete(verifyStatus(ctx, HTTP_CREATED));
+
+    doPost(client, resourceUrl(), pojo2JsonObject(library))
```

Comment:
There you are doing 2 posts, that will work in parallel. So, there could be a case when second post will be completed before 1st one and test will fail.
You should do 2nd post after 1st is completed.

Reply:
Good point, Fixed